thewalrus/symplectic.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Symplectic
+==========
+
+**Module name:** :mod:`thewalrus.symplectic`
+
+.. currentmodule:: thewalrus.symplectic
+
+Contains some Gaussian operations and auxiliary functions.
+
+Auxiliary functions
+-------------------
+
+.. autosummary::
+    expand
+    reduced_state
+
+Gaussian states
+---------------
+
+.. autosummary::
+    vacuum_state
+
+
+Gates and operations
+--------------------
+
+.. autosummary::
+    two_mode_squeezing
+    interferometer
+    loss
+    mean_photon_number
+    beam_splitter
+    rotation
```

Comment:
Do we want to add _all_ the missing gates and states above?

Reply:
What do you mean by *all*?

Reply:
oops, didn't mean to italicize. I meant should we add all symplectics that aren't currently provided, like Sgate, Coherent, CX, CZ, etc. I suppose we should do this in a new PR.

Reply:
Sgate is there. As for the others I don't think it is worth it.